mpisppy/utils/config.py

```diff
@@ -625,33 +625,46 @@ def gradient_args(self):
                            description="name of the gradient rho file (must be csv)",
                            domain=str,
                            default='')
-        self.add_to_config("order_stat",
-                           description="order statistic for rho (must be between 0 and 1)",
+        self.add_to_config("grad_display_rho",
+                           description="display rho during gradient calscs (default True)",
+                           domain=bool,
+                           default=True)
+        self.add_to_config("grad_primal_thresh",
+                           description="primal threshold for diff during gradient calcs",
+                           domain=float,
+                           default=0.001)
+        self.add_to_config("grad_dual_thresh",
+                           description="dual threshold for abs norm during gradient calcs",
+                           domain=float,
+                           default=0.1)
+        self.add_to_config("grad_pd_thresh",
+                           description="threshold for dual/primal during gradient calcs",
+                           domain=float,
+                           default=0.1)
+        self.add_to_config("grad_order_stat",
+                           description="order statistic for gradient based rho (must be between 0 and 1)",
                            domain=float,
                            default=-1.0)
```

Comment:
Shouldn't the default value for the order statistic be valid? (as in on the interval (0,1))?

Reply:
He wants to require value be given, so he flagged it. The default should be None, really (but then the test for a missing entry where the value is used needs to be changed). The alternative is to set the default to 0.5 and drop the test for a missing entry. If I get time, I will do the former; but I won't get time until tomorrow at the earliest.

Reply:
done